presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java

```diff
@@ -306,6 +328,14 @@ public QueryInfo createQuery(Session session, String query)
             // This is intentionally not a method, since after the state change listener is registered
             // it's not safe to do any of this, and we had bugs before where people reused this code in a method
             URI self = locationFactory.createQueryLocation(queryId);
+
+            // if session creation failed, create a minimal session object
+            if (session == null) {
+                session = Session.builder(new SessionPropertyManager())
```

Comment:
[Issue]
This seems like a fake session object without full capabilities, but cannot be recognised as such in any way other than context.
Especially using `new SessionPropertyManager()` seems like a source of future problems with this Session object usage. Cannot we use injected sessionPropertyManager instead?